rules/Transform/Rector/Attribute/AttributeKeyToClassConstFetchRector.php

```diff
@@ -151,19 +150,27 @@ private function processToClassConstFetch(
 
                 $constName = $attributeKeyToClassConstFetch->getValuesToConstantsMap()[$value] ?? null;
                 if ($constName === null) {
-                    continue;
+                    return false;
                 }
 
-                $arg->value = $this->nodeFactory->createClassConstFetch(
+                $newValue = $this->nodeFactory->createClassConstFetch(
                     $attributeKeyToClassConstFetch->getConstantClass(),
                     $constName
                 );
 
-                $hasChanged = true;
-                continue 2;
+                if (
+                    $arg->value instanceof Node\Expr\ClassConstFetch
+                    && $this->getName($arg->value) === $this->getName($newValue)
+                ) {
+                    return false;
+                }
+
+                $arg->value = $newValue;
+
+                return true;
```

Comment:
when we use `$hasChanged` assigned, no need to return `true`, only back to use `$hasChanged` return after loop.